stdlib/public/core/Hashing.swift

```diff
@@ -203,7 +203,13 @@ func _squeezeHashValue(_ hashValue: Int, _ upperBound: Int) -> Int {
 @_transparent
 public // @testable
 func _combineHashValues(_ firstValue: Int, _ secondValue: Int) -> Int {
-  let magic = 0x9e3779b9 as UInt // Based on the golden ratio.
+  // Use a magic number based on the golden ratio
+  // (0x1.9e3779b97f4a7c15f39cc0605cedc8341082276bf3a27251f86c6a11d0c18e95p0).
+#if arch(i386) || arch(arm)
+  let magic = 0x9e3779b9 as UInt
+#elseif arch(x86_64) || arch(arm64) || arch(powerpc64) || arch(powerpc64le) || arch(s390x)
+  let magic = 0x9e3779b97f4a7c15 as UInt
+#endif
```

Comment:
Would it make sense to have compiler directives like `#if archbits(64) ... #elseif archbits(32) ... #endif`? Seems problematic/error prone to list individual architectures.

Reply:
We can already write this as `if MemoryLayout<Int>.size == 4`, etc., and the compiler takes care of optimizing it. But here I'm following the prevailing style in this particular file.

Reply:
Is it smart enough not to complain that the 64-bit value is out of bounds on 32-bit platforms if you write that?

I'm not critiquing what you did here — more just wondering how this could be more clear. Totally possible that this is basically only an issue in places like this in the standard library.

Reply:
That’s a good point; the code may not actually compile on a 32-bit system depending on compiler smarts.

I think, ultimately, the utility of another compilation conditional will come down to whether or not we realistically expect Swift to support 32–bit architectures other than i386 and arm. It’s not entirely out of the question but does seem pretty unlikely in the near term.

Reply:
A dedicated directive would certainly express the underlying intention better, even if we never add support for additional architectures. We already have `_endian(little)`/`_endian(big)` for endianness.

In any case, for now, I'd prefer to replace the `#elseif` with `#else` here, and rely on the compiler to complain if the constant doesn't fit in an `UInt`. (It seems unnecessarily fragile/verbose to list all supported 64-bit architectures, although it is arguably the right choice in the rest of the file. We did go with `#else` in the new `String` implementation, although that may have just been due to laziness on my part. 🙈)

Reply:
I agree that it's a pragmatic solution for the time being. I'll create a follow-up PR with that change.